tests/test_views_admin.py

```diff
@@ -0,0 +1,242 @@
+        )
+        self.app.put_json(
+            self.safe_group_containing_doomed_user,
+            {"data": {"members": [self.doomed_user_principal, self.user_principal]}},
+            headers=self.headers,
+        )
+
+        # Actually do the delete. Each `test_` method checks some
+        # consequence of what got deleted.
+        self.app.delete(self.delete_user_url, headers=self.headers)
+
+    def test_doomed_bucket_was_deleted(self):
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("bucket", "", "user_to_delete")
+
+    def test_doomed_bucket_collection_was_deleted(self):
+        # Everything under this bucket should be gone
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("collection", self.doomed_bucket_url, "collection")
+
+    def test_doomed_bucket_doomed_collection_was_deleted(self):
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("collection", self.doomed_bucket_url, "collection_to_delete")
+
+    def test_safe_bucket_was_not_deleted(self):
+        self.storage.get("bucket", "", "user_safe")
+
+    def test_safe_bucket_doomed_collection_was_deleted(self):
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("collection", self.safe_bucket_url, "collection_to_delete")
+
+    def test_safe_collection_was_not_deleted(self):
+        self.storage.get("collection", self.safe_bucket_url, "collection_safe")
+
+    def test_safe_collection_safe_record_was_not_deleted(self):
+        self.storage.get("record", self.safe_collection_url, "record_safe")
+
+    def test_safe_collection_doomed_record_was_deleted(self):
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("record", self.safe_collection_url, "record_to_delete")
+
+    def test_safe_group_doomed_user_was_removed(self):
+        principals = self.permission.get_user_principals(self.doomed_user_principal)
+        for principal in principals:
+            if principal.startswith("system:"):
+                continue
+            if principal == self.doomed_user_principal:
+                continue
+            self.fail(f"Got a leftover principal: {principal}")
+
+    def test_doomed_collection_no_longer_has_any_perms(self):
+        permissions = self.permission.get_object_permissions(self.doomed_collection_url)
+        self.assertEqual(permissions, {})
+
+    def test_doomed_bucket_collection_no_longer_has_any_perms(self):
+        permissions = self.permission.get_object_permissions(self.doomed_bucket_collection_url)
+        self.assertEqual(permissions, {})
+
+    def test_safe_bucket_no_longer_has_doomed_user_perm(self):
+        permissions = self.permission.get_object_permissions(self.safe_bucket_url)
+        for perm, principals in permissions.items():
+            self.assertNotIn(self.doomed_user_principal, principals)
```

Comment:
Assert that we still have `self.user_principal` ?

(permissions should not be empty)